src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -858,6 +858,17 @@ def __init__(self) -> None:
         self.message = ""  # Freeform description
         self.is_paused: bool = False  # Is the upgrade paused?
 
+    def to_json(self) -> dict:
+        return {
+            'in_progress': self.in_progress,
+            'target_image': self.target_image,
+            'which': self.which,
+            'services_complete': self.services_complete,
+            'progress': self.progress,
+            'message': self.message,
+            'is_paused': self.is_paused,
+        }
```

Comment:
@adk3798 could you do a quick review here?

Reply:
LGTM